ravenframework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -103,125 +118,251 @@ def _evaluate(self, datasets, **kwargs):
+          regrExp = OrthogonalMatchingPursuit(fit_intercept=False).fit(yExpStd, yAppStd)
+          yExpReg = regrExp.predict(yExpStd)
+          # Combine measurements by multiple Experiment regression
+          yMsrReg = regrExp.predict(yMsrStd)
+
+        # Measurement PDF with KDE
+        knlMsr = stats.gaussian_kde(yMsrReg)
+
+        # KDE for joint PDF between Exp and App
+        m1 = yExpReg[:]
+        m2 = yAppStd.flatten()
+        xmin = m1.min()
+        xmax = m1.max()
+        ymin = m2.min()
+        ymax = m2.max()
+        # Grid of Experiment (X), grid of Application (Y)
+        X, Y = np.mgrid[xmin:xmax:self.binKDE, ymin:ymax:self.binKDE]
+        psts = np.vstack([X.ravel(), Y.ravel()])
+        vals = np.vstack([m1, m2])
+        # Measurement PDF over Exp range
+        pdfMsr = knlMsr(X[:, 0])
+
+        # Condition number of matrix of feature and target
+        condNum = np.linalg.cond(vals)
+        # If condition number is greater than 100
+        invErr = 100
+        # Check whether the covavariance matrix is positive definite
+        if condNum>=invErr:
+          # If singular matrix, measurement of Experiment is directly transfered
+          # as predicted Application
+          pdfAppPred = knlMsr(Y[0, :])
+        else:
+          # If not, KDE of Experiment and Application
+          knl = stats.gaussian_kde(vals)
+          # Joint PDF of Experiment and Application
+          Z = np.reshape(knl(psts).T, X.shape)
+          # yAppPred by integrating p(yexp, yapp)p(ymsr) over [yexp.min(), yexp.max()]
+          pdfAppPred = np.dot(Z, pdfMsr.reshape(pdfMsr.shape[0], 1))
+
+        # Normalized PDF of predicted application
+        pdfAppPredNorm = pdfAppPred.flatten()/pdfAppPred.sum()/np.diff(Y[0, :])[0]
+
+        # Calculate Expectation (average value) of predicted application
+        # by integrating xf(x), where f(x) is PDF of x
+        predMean = 0.0
+        for i in range(len(Y[0, :])):
+          predMean += Y[0, i]*pdfAppPredNorm[i]*(Y[0, 1]-Y[0, 0])
+
+        # Calculate Variance of predicted application
+        # by integrating (x-mu_x)^2f(x), where f(x) is PDF of x
+        predVar = 0.0
+        for i in range(len(Y[0, :])):
+          predVar += (Y[0, i]-predMean)**2.0 * pdfAppPredNorm[i]*(Y[0, 1]-Y[0, 0])
+
+        # Predicted standard deviation is square root of variance
+        predStd = np.sqrt(predVar)
+        # Prior standard deviation is the sample standard deviation
+        # Consider probability weights in the future
+        priStd = np.std(yAppStd)
+        # Uncertainty reduction fraction is 1.0-sigma_pred/sigma_pri
+        name = "pri_post_stdReduct_" + targ.split('|')[-1]
+        outputDict[name] = (1.0-predStd/priStd)
+
+    # Snapshot PCM
+    if PCMVersion == 'Snapshot':
+      outputArray = []
 
-      # Single Experiment response
-      if yExpStd.shape[1]==1:
+      for feat, msr, targ in zip(self.features, self.measurements, self.targets):
```

Comment:
why self.targets only appears in snapshot method, not both?

Reply:
> I have several comments that I have added in the PR. In addition, please check the following:
> 
> 1. check all the csv files, if they are not used, please remove the file
> 2. It seems there is no results for snapshot test
> 3. You need to update tests file in your test folder in order your added tests can be tested. (I can show you how to do it.) Here is the example:
>    ![image](https://user-images.githubusercontent.com/7321071/211610591-a8b1b47d-8103-4cb2-8c30-3cd2c9cde36c.png)
> 
> The csv files need to be placed in the gold folder ![image](https://user-images.githubusercontent.com/7321071/211610767-83f6384e-6e08-4195-aff8-0cf1e8731540.png)

@wangcj05  Thanks for the comments Congjian. According to the suggestions: 
1. The unused data in StaticExample folder has been removed. The 877 csv files in SnapshotExample folder are are the 877 samples and all needed for test.
2. The results for snapshot test are in file 'pcmSnapshot_metric_dump_0.csv', which has been moved to gold folder.
3. The 'tests' file has been updated with static/snapshot PCM test info.

Reply:
> why self.targets only appears in snapshot method, not both?

It has been included in static method previously. Now I have simplified the _evaluate function and combine the data reading for static/snapshot PCM.